python/src/scipp/_shape.py

```diff
@@ -71,6 +102,36 @@ def fold(x, dim, sizes=None, dims=None, shape=None):
     :raises: If the volume of the old shape is not equal to the
              volume of the new shape.
     :return: Variable or DataArray with requested dimension labels and shape.
+
+    Examples:
+
+      >>> v = sc.arange('x', 6)
+      >>> v
+      <scipp.Variable> (x: 6)      int64  [dimensionless]  [0, 1, ..., 4, 5]
+      >>> sc.fold(v, 'x', {'y': 2, 'z': 3})
```

Comment:
Use keyword args everywhere?

Reply:
These arguments cannot be confused because they all ave different types. And I find this annoying to type already.
Do we want to encourage always using keywords or only when they can avoid confusion?

Reply:
I think that is what we had a greed on a couple month ago? Always encourage keyword args. In fact in many cases we made keyword args *mandatory* already.

Reply:
Is `sc.concatenate(x=a, y=b, dim='x')` more readable / meaningful? I am fine with putting the `dim` kwarg, but `x` and `y` don't help at all in my view.

Reply:
Agree about `x` and `y`.